Dockerfile

```diff
@@ -1,108 +1,193 @@
+#
+# Install system requirements.
+# We update, upgrade, and delete lists all in one layer
+# in order to reduce total image size.
 RUN apt-get update && \
-    # Global requirements
     DEBIAN_FRONTEND=noninteractive apt-get install --yes \
-    build-essential \
-    curl \
-    # If we don't need gcc, we should remove it.
-    g++ \
-    gcc \
-    git \
-    git-core \
-    language-pack-en \
-    libfreetype6-dev \
-    libmysqlclient-dev \
-    libssl-dev \
-    libxml2-dev \
-    libxmlsec1-dev \
-    libxslt1-dev \
-    swig \
-    # openedx requirements
-    gettext \
-    gfortran \
-    graphviz \
-    libffi-dev \
-    libfreetype6-dev \
-    libgeos-dev \
-    libgraphviz-dev \
-    libjpeg8-dev \
-    liblapack-dev \
-    libpng-dev \
-    libsqlite3-dev \
-    libxml2-dev \
-    libxmlsec1-dev \
-    libxslt1-dev \
-    # lynx: Required by https://github.com/edx/edx-platform/blob/b489a4ecb122/openedx/core/lib/html_to_text.py#L16
-    lynx \
-    ntp \
-    pkg-config \
-    python3-dev \
-    python3-venv \
-    && rm -rf /var/lib/apt/lists/*
+        # Global requirements
+        build-essential \
+        curl \
+        # If we don't need gcc, we should remove it.
+        g++ \
+        gcc \
+        git \
+        git-core \
+        language-pack-en \
+        libfreetype6-dev \
+        libmysqlclient-dev \
+        libssl-dev \
+        libxml2-dev \
+        libxmlsec1-dev \
+        libxslt1-dev \
+        swig \
+        # openedx requirements
+        gettext \
+        gfortran \
+        graphviz \
+        libffi-dev \
+        libfreetype6-dev \
+        libgeos-dev \
+        libgraphviz-dev \
+        libjpeg8-dev \
+        liblapack-dev \
+        libpng-dev \
+        libsqlite3-dev \
+        libxml2-dev \
+        libxmlsec1-dev \
+        libxslt1-dev \
+        # lynx: Required by https://github.com/edx/edx-platform/blob/b489a4ecb122/openedx/core/lib/html_to_text.py#L16
+        lynx \
+        ntp \
+        pkg-config \
+        python3-dev \
+        python3-venv && \
+    rm -rf /var/lib/apt/lists/*
 
+# Set locale.
 RUN locale-gen en_US.UTF-8
-ENV LANG en_US.UTF-8
-ENV LANGUAGE en_US:en
-ENV LC_ALL en_US.UTF-8
 
-WORKDIR /edx/app/edxapp/edx-platform
+# Env vars: locale
+ENV LANG='en_US.UTF-8'
+ENV LANGUAGE='en_US:en'
+ENV LC_ALL='en_US.UTF-8'
 
-ENV PATH /edx/app/edxapp/nodeenv/bin:${PATH}
-ENV PATH ./node_modules/.bin:${PATH}
-ENV CONFIG_ROOT /edx/etc/
-ENV PATH /edx/app/edxapp/edx-platform/bin:${PATH}
-ENV SETTINGS production
-RUN mkdir -p /edx/etc/
+# Env vars: configuration
+ENV CONFIG_ROOT='/edx/etc'
+ENV LMS_CFG="$CONFIG_ROOT/lms.yml"
+ENV STUDIO_CFG="$CONFIG_ROOT/studio.yml"
+ENV EDX_PLATFORM_SETTINGS='production'
 
-ENV VIRTUAL_ENV=/edx/app/edxapp/venvs/edxapp
-RUN python3.8 -m venv $VIRTUAL_ENV
+# Env vars: path
+ENV VIRTUAL_ENV='/edx/app/edxapp/venvs/edxapp'
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ENV PATH="./node_modules/.bin:${PATH}"
```

Comment:
Hm, the original Dockerfile used a relative path. I'm going to dig around git blame to see if there's any reason for using relative instead of absolute.

Reply:
Coming up dry on any reasoning for the relative path. Since the `edxops/edxapp` image uses the full path, I'll update this to also use the full path. Thanks for catching.